uptane-standard.md

```diff
@@ -240,28 +277,222 @@ The Time Server SHALL expose a public interface allowing primaries to communicat
 
 ## In-vehicle implementation requirements
 
+An Uptane-compliant ECU SHALL be able to download and verify the time, metadata, and image binaries before installing a new image.
+
+Each ECU in a vehicle receiving over-the-air updates is either a primary or a secondary ECU. A primary ECU collects and delivers to the Director vehicle manifests ({{vehicle_version_manifest}}) containing information about which images have been installed on ECUs in the vehicle. It also downloads and verifies the latest time, metadata, and images for itself and for its secondaries. A secondary ECU downloads and verifies the latest time, metadata, and images for itself from its associated primary ECU. It also sends signed information about its installed images to its associated primary.
+
+All ECUs MUST verify image metadata as specified in {{metadata_verification}} before installing an image or making it available to other ECUs. A primary ECU MUST perform full verification ({{full_verification}}). A secondary ECU SHOULD perform full verification if possible, and MUST perform full verification if it is safety-critical. If it is not safety-critical, it MAY perform partial verification ({{partial_verification}}) instead.
```

Comment:
I'm worried that "and MUST perform full verification if it is safety-critical" is too vague to be helpful in a spec. As I see it, there are two primary concerns:

1. How do we define "safety-critical"? It seems unreasonable to make something a "MUST" if we don't define the conditional. There are some formal definitions for safety-critical systems, but we should explicitly reference them if we want to use such definitions.
2. Is safety really the only case where a secondary ECU MUST perform full verification? For example, there are some infotainment designs which aren't safety-critical, but have **very** high privacy or financial impacts if compromised.

Reply:
Regarding (1.), I think referencing an external definition is a good idea. Can you suggest one? As for (2.), I don’t really see any good alternative: we already say that all ECUs SHOULD do full verification; trying to define another MUST runs into the problem of definition that you raise in (1.).

Reply:
Regarding (1): There's this book which has some definitions: https://www.sae.org/publications/books/content/pt-103/ (I don't recall how useful they were). I'm going to chat with some safety people at work and see if they know of a canonical definition in automotive.

Regarding (2): I don't either. I guess my comment was hinting that maybe we just shouldn't include "MUST" at all for this case. I'm not hard-set on that, however.

Reply:
I guess A.3.2.10 from https://www.system-safety.org/Documents/MIL-STD-882D.pdf would work as a particular definition. It's not as precise as I'd like, but it's a start and something that can be referenced.

Reply:
Sounds reasonable. I'll change the language to something like "A secondary ECU SHOULD perform full verification if possible. See Uptane Deployment Guidelines for considerations that should be taken for choosing full or partial verification."